### PR TITLE
Support purging config settings

### PR DESCRIPTION
**Pull request by alexjfisher**
Support removing config with `ensure => absent`.  In a separate commit I've converted the class to use puppet 4 style parameter validation.

### CODE REVIEW

File: manifests/config.pp [hunk 1]
```diff
@@ -1,39 +1,37 @@
...
   }
 
   file_line { "powerdns-config-${setting}-${path}":
-    ensure  => $ensure,
-    path    => $path,
-    line    => $line,
-    match   => "^${setting}=",
-    require => Package[$require_package],
-    notify  => Service[$notify_service],
-    before  => Service[$notify_service],
```
Comment (alexjfisher) [inline manifests/config.pp]: `notify` implies `before` so `before` is not needed.

File: manifests/config.pp [hunk 2]
```diff
@@ -1,39 +1,37 @@
...
+    ensure            => $ensure,
+    path              => $path,
+    line              => $line,
+    match             => "^${setting}=",
+    match_for_absence => true, # ignored when ensure == 'present'
```
Comment (alexjfisher) [inline manifests/config.pp]: https://github.com/puppetlabs/puppetlabs-stdlib/blob/cef3e894d642207d2f87e307159b90ce49fcd455/lib/puppet/type/file_line.rb#L110

**Comment by ju5t** [comment]
I completely missed that it passes all checks now. Thanks for this @alexjfisher!

**Comment by alexjfisher** [comment]
@ju5t And I forgot all about this too!  Thanks for merging.

**Pull request merged**